packages/tracing/src/integrations/mysql.ts

```diff
@@ -0,0 +1,75 @@
+import { Hub } from '@sentry/hub';
+import { EventProcessor, Integration } from '@sentry/types';
+import { dynamicRequire, fill, logger } from '@sentry/utils';
+
+interface MysqlConnection {
+  end: () => void;
+  prototype: {
+    query: () => void;
+  };
+}
+
+/** Tracing integration for node-mysql package */
+export class Mysql implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Mysql';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = Mysql.id;
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    let connection: MysqlConnection;
+
+    try {
+      const mysqlModule = dynamicRequire(module, 'mysql') as {
+        createConnection: (options: unknown) => MysqlConnection;
+      };
+      const conn = mysqlModule.createConnection({});
+      connection = (conn.constructor as unknown) as MysqlConnection;
+      conn.end();
+    } catch (e) {
+      logger.error('Mysql Integration was unable to require `mysql` package.');
+      return;
+    }
+
+    /**
+     * function (query, callback) => void
+     * function (query, params, callback) => void
+     * function (query) => Promise
+     * function (query, params) => Promise
+     */
```

Comment:
Any chance these could be `//` instead? It took me a good 5 sec to realize this wasn't the start of a new function.